### ISSUE TITLE
bulkCreate with updateOnDuplicate broken when field names different than column names.

### ISSUE THREAD

**Issue by bomattin** [sequelize/sequelize#4600]
Dialect: mysql
Sequelize Version: 3.9.0

I've got an Event model:

```
id: {
   ...
   field: "event_id"
},
group: {
   ...
   field: "group_id"
},
start: {
   ...
   field: "start_date"
},
end: {
   ...
   field: "end_date"
},
content: {
   ...
   field: "content"
},
type: {
   ...
   field: "type"
} 
```

Notice that some fields of this object have different corresponding column names in the table they're stored in.

Now when I try to do a bulkUpdate with the following code: 

```
models.Event.bulkCreate(uEvents, {
        fields: ['id', 'start', 'end', 'type', 'group', 'content', 'title', 'updatedAt'],
        updateOnDuplicate: ["start_date", "end_date", "type", "group_id", "content", "title", "updatedAt"]
    })
```

Sequelize generates the following SQL:

```
INSERT INTO `events` (`event_id`,`start_date`,`end_date`,`type`,`group_id`,`content`,`title`,`updatedAt`,`createdAt`) 
	VALUES (4,'2016-04-01 00:00:00','2016-06-30 00:00:00','range',6,'Directory Launch Phase II','Maintenance and Feature Addition','2015-09-18 16:43:25','2015-10-01 20:32:14'),
		   (3,'2016-01-01 00:00:00','2016-04-01 00:00:00','range',6,'Directory Launch Phase I','Initial Launch for Directory and API','2015-09-18 16:42:52','2015-10-01 20:32:14') 

	ON DUPLICATE KEY UPDATE `content`=VALUES(`content`),`title`=VALUES(`title`),`type`=VALUES(`type`),`updatedAt`=VALUES(`updatedAt`);
```

Notice that the only fields included in the UPDATE clause are those who share their name with their table column.

Ok, so what if we try identifying the `updateOnDuplicate` fields to their property names, e.g.:

`updateOnDuplicate: ["id", "start", ...]`

No dice. We get SQL that looks like this:

```
ON DUPLICATE KEY UPDATE `group`=VALUES(`group`),`start`=VALUES(`start`),`end`=VALUES(`end`),`content`=VALUES(`content`),`title`=VALUES(`title`),`type`=VALUES(`type`),`updatedAt`=VALUES(`updatedAt`)
```

which results in an `ER_BAD_FIELD_ERROR` since the *column* names are never aliased as *table* names, and the database knows to insert into `group_id`, but doesn't know what the hell `group` is.

In short: The `options.fields` and `options.updateOnDuplicate` arguments are using different ways of referring to field names, and there's no way to make them line up (that I'm aware of).
